lib/matplotlib/axes/_base.py

```diff
@@ -3240,10 +3240,11 @@ def set_xlim(self, left=None, right=None, emit=True, auto=False,
         left = self._validate_converted_limits(left, self.convert_xunits)
         right = self._validate_converted_limits(right, self.convert_xunits)
 
+        old_left, old_right = self.get_xlim()
+
         if left is None or right is None:
             # Axes init calls set_xlim(0, 1) before get_xlim() can be called,
             # so only grab the limits if we really need them.
```

Comment:
Given the comment here, I'd assume that this actually needs to stay here (... for init order purposes) and you need to duplicate the call to get_xlim() under `if self.get_xscale("log"): ...`?

Reply:
Oops yes, I've fixed that